data/json/items/gun/faults_gun.json

```
@@ -2,13 +2,26 @@
   {
     "id": "fault_gun_blackpowder",
     "type": "fault",
-    "name": "Blackpowder fouling",
-    "description": "Firing blackpowder loads from a gun fouls it, which reduces reliability and, if left uncleaned, leads to rust.  It fouls the gun much faster than the use of modern smokeless powder cartridges.  Fouling is only a significant impact on reliability at high levels, but black powder fouling accumulates quickly.",
-    "time": "50 m",
+    "name": "Clean blackpowder fouling without lubricating",
```

```suggestion
    "name": "Clean blackpowder fouling without lubricating",
```
This is too hacky. The fault name is also displayed in the item info screen, and this name would not make much sense there. I'd suggest you keep the original name, and change the mending menu code to display the options to clean with or wihout lubricant instead.

---

Kevin didn't want the optional recipe component approach for one mending action solution.

---

It doesn't have to be an optional component. It can be two separate recipes that results in different outcomes. Or was he also against multiple recipes for a single mending action?

---

You mean one mending action with, for example, one component line that reads "motor oil OR gasoline" with only the former resulting in lubrication and the latter curing the dirty condition but creating the unlubricated fault??

---

Nope, that's still the optional recipe component approach. I mean you can change the current format `"requirements": { <requirements>}` to something like this:
```json
"requirements: [
  {
    "name": "clean without lubricating",
    "turns_into": "fault_unlubricated",
    <requirements A>
  },
  {
    "name": "clean and lubricate",
    <requirements B>
  }
]
```
So the player can choose to mend the fault with either "clean without lubricating", which turns the fault into "fault_unlubricated", or "clean and lubricate", which completely removes the fault. Of course, currently it does not support this format, so you'll have to implement it in the code.

---

So you mean there would still be multiple different faults. The only difference would be how new faults are added: not hardcoded, but applied to all fault removals if there is a "turns_into" parameter for the fault?

---

Mostly yes, but "fault_gun_dirt_and_lube" should also be removed and re-implemented with the format in my previous comment.

---

Are you talking about adding a copy-from parameter system to faults?

---

No, I mean "fault_gun_dirt_and_lube" shouldn't be a separate fault: It's in every way the same fault as "fault_gun_dirt", the only difference is the method to mend the fault. So what really need to be done is not adding "fault_gun_dirt_and_lube" as a new separate fault, but changing the previous mending recipe of "fault_gun_dirt" to two separate recipes with different requirements and results. Also, as I said, the name of the fault shouldn't be changed to the name of the mending method itself, since the name is expected to be a description of the fault itself in the item info display. I mean, it does not make sense to say your gun has a "clean blackpowder fouling without lubricating" problem, when what it really has is a "blackpowder fouling" problem.

---

> in

I have no clue how to do a major rewrite like that. The best solution I can think of is always requiring some sort of lubricant to clean a gun and scrapping the unlubricated condition and it's effects altogether (as it would no longer be possible to clean a gun without lubricating it). "Fast-cleaning" a gun is already in this PR in the form of reloading when dirt level is high, which in exchange for time slightly reduces dirt in the sense of some heavy dirt accumulation falling out of the gun.

Or we could always add unlubricated after cleaning, but drastically reduce time under the assumption that lubricating is easier than cleaning and that the character relies on the moving of the gun parts during firing to adequately spread the oil.

---

I'm leaning towards the second option. Once we have the infrastructure to support multiple mending recipes we can then add the clean-and-lubricate action back.

---

Yeah, I think that's what I'll do. Many guns, esp. modern handguns just have a level you flip or a button you push and the gun is disassembled in seconds. Obviously not the case with ALL guns, but lubricating is clearly much faster to perform than cleaning at the very least because there's no need to lay out and assemble the rods and bore brushes.

---

That ought to do it.

---

> Yeah, I think that's what I'll do. Many guns, esp. modern handguns just have a level you flip or a button you push and the gun is disassembled in seconds. Obviously not the case with ALL guns, but lubricating is clearly much faster to perform than cleaning at the very least because there's no need to lay out and assemble the rods and bore brushes.

Implemented.

---

> I'm leaning towards the second option. Once we have the infrastructure to support multiple mending recipes we can then add the clean-and-lubricate action back.

Implemented.